### ISSUE TITLE
Implement support for shared formulas

### ISSUE THREAD

**Issue by Shimuuar** [qrilka/xlsx#101]
I'm speaking about feature like:

```
    <row r="2" spans="1:3" x14ac:dyDescent="0.25">
      <c r="C2" t="str">
        <f>IF(ISTEXT(A2), "string", IF(ISNUMBER(A2), "Number", IF(ISLOGICAL(A2), "bool", "-")))</f>
        <v>-</v>
      </c>
    </row>
    <row r="3" spans="1:3" x14ac:dyDescent="0.25">
      <c r="C3" t="str">
        <f t="shared" ref="C3:C8" si="0">IF(ISTEXT(A3), "string", IF(ISNUMBER(A3), "Number", IF(ISLOGICAL(A3), "bool", "-")))</f>
        <v>-</v>
      </c>
    </row>
    <row r="4" spans="1:3" x14ac:dyDescent="0.25">
      <c r="C4" t="str">
        <f t="shared" si="0"/>
        <v>-</v>
      </c>
```

Looks like it's known problem as haddock have

> TODO: array. dataTable and shared formula types support

**Comment by qrilka** [comment]
@Shimuuar probably you have some real life example file from Excel?

**Comment by Shimuuar** [comment]
Here it is: [Work+Document+Anon2.xlsx](https://github.com/qrilka/xlsx/files/1338338/Work.Document.Anon2.xlsx)

It's fairly rich in features and conveniently large.

**Comment by Shimuuar** [comment]
Here it is: [expr-import.xlsx](https://github.com/qrilka/xlsx/files/1338349/expr-import.xlsx)

It's stripped down example

**Comment by qrilka** [comment]
Resolved by d40bfb2
